frontend/model/contracts/group.js

```diff
@@ -1147,6 +1147,31 @@ sbp('chelonia/defineContract', {
         if (!state.generalChatRoomId) {
           Vue.set(state, 'generalChatRoomId', data.chatRoomID)
         }
+      },
+      async sideEffect ({ data, meta, contractID }, { state: Rstate }) {
+        const rootState = sbp('state/vuex/state')
+        const contracts = rootState.contracts || {}
+        const { identityContractID } = rootState.loggedIn
+        const userState = rootState[identityContractID]
+        const state = rootState[contractID]
+
+        if (rootState[data.chatRoomID]?._volatile) {
+          return
+        }
+
+        await sbp('chelonia/out/keyRequest', {
+          originatingContractID: identityContractID,
+          originatingContractName: contracts[identityContractID].type,
+          contractID: data.chatRoomID,
+          contractName: 'gi.contracts/chatroom',
+          signingKey: state._volatile?.keys?.[(((Object.values(Object(state._vm?.authorizedKeys)): any): GIKey[]).find((k) => k?.meta?.type === 'csk')?.id: ?string)],
```

Comment:
Which key is being used here? It seems to say "pick the first CSK of the authorizedKeys"? Is that right? If so, why would that be correct? If not, what is going on?

Reply:
That's correct: it does "pick the first CSK of the authorizedKeys". Not sure what you mean by 'why would that be correct?' though. What scenarios do you have in mind that would _not_ make this correct? If it's the "pick the first CSK" part that's confusing, you can read it as "pick the ~~first~~ CSK", as there should be only one CSK at any given time.